sphinx/tutorial/cylc/furthertopics/inheritance.rst

```diff
@@ -0,0 +1,603 @@
+       title = Airbus A380 Jumbo-Jet                   # Defined in a380
+   [[[environment]]]
+       CAN_TAKE_OFF_VERTICALLY = false                 # Inherited from AIRPLANE
+
+Note that the ``description`` setting is defined in the ``AIR_VEHICLE``
+family but is overwritten by the value specified in the ``AIRPLANE`` family.
+
+
+Multiple Inheritance
+--------------------
+
+Next we want to add a vehicle called the V-22 Osprey to the suite. The V-22
+is a cross between a plane and a helicopter - it has wings but can take-off and
+land vertically.
+
+.. image:: https://upload.wikimedia.org/wikipedia/commons/e/e3/MV-22_mcas_Miramar_2014.JPG
+   :width: 300px
+   :align: center
+
+As the V-22 can be thought of as both a plane and a helicopter we want it to
+inherit from both the ``AIRPLANE`` and ``HELICOPTER`` families. In cylc we can
+inherit from multiple families by separating their names with commas:
+
+Add the following task to your ``suite.rc`` file.
+
+.. code-block:: cylc
+
+       [[v22]]
+           inherit = AIRPLANE, HELICOPTER
+           [[[meta]]]
+               title = V22 Osprey military aircraft.
+
+Refresh your cylc graph window or re-run the cylc graph command.
+
+The inheritance hierarchy should now look like this:
+
+.. digraph:: Example
+   :align: center
+
+   bgcolor=none
+
+   AIRPLANE  [color=royalblue,
+      fillcolor=powderblue,
+      shape=box,
+      style=filled];
+   v22    [color=royalblue,
+      fillcolor=powderblue,
+      shape=box,
+      style=filled];
+   AIRPLANE -> v22    [color=royalblue];
+   a380   [color=royalblue,
+      fillcolor=powderblue,
+      shape=box,
+      style=filled];
+   AIRPLANE -> a380   [color=royalblue];
+   HELICOPTER   [color=royalblue,
+      fillcolor=powderblue,
+      shape=box,
+      style=filled];
+   HELICOPTER -> v22  [color=royalblue];
+   r44    [color=royalblue,
+      fillcolor=powderblue,
+      shape=box,
+      style=filled];
+   HELICOPTER -> r44  [color=royalblue];
+   root   [color=royalblue,
+      fillcolor=powderblue,
+      shape=box,
+      style=filled];
+   VEHICLE   [color=royalblue,
+      fillcolor=powderblue,
+      shape=box,
+      style=filled];
+   root -> VEHICLE    [color=royalblue];
+   AIR_VEHICLE  [color=royalblue,
+      fillcolor=powderblue,
+      shape=box,
+      style=filled];
+   VEHICLE -> AIR_VEHICLE   [color=royalblue];
+   AIR_VEHICLE -> AIRPLANE  [color=royalblue];
+   AIR_VEHICLE -> HELICOPTER   [color=royalblue];
+
+Inspect the configuration of the ``v22`` task using the ``cylc get-config``
+command.
+
+.. spoiler:: Hint warning
+
+   .. code-block:: bash
+
+      cylc get-config . --sparse -i "[runtime][v22]"
+
+You should see that the ``CAN_TASK_OFF_VERTICALLY`` environment variable has
+been set to ``false`` which isn't right. This is because of the order in which
+inheritance is applied.
+
+Cylc handles multiple-inheritance by applying each family from last to first.
```

Comment:
Clarifying that 'last to first' is implying right to left as assigned to the inheritance variable (as suggested but not fully elucidated in the previous sentence) might be a good idea, particularly as 'first' and 'last' are also used to describe the application order e.g. HELICOPTER is *applied* first but *specified as a family* last. This is in fact implicit from lines 260-267 but best be explicit.